pkg/operation/botanist/namespaces.go

```diff
@@ -103,3 +116,31 @@ func (b *Botanist) WaitUntilSeedNamespaceDeleted(ctx context.Context) error {
 func (b *Botanist) DefaultShootNamespaces() component.DeployWaiter {
 	return namespaces.New(b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 }
+
+// getShootExtensionTypes returns all extension types that are enabled or explicitly disabled for the shoot.
+func getShootExtensionTypes(b *Botanist, ctx context.Context) (map[string]bool, error) {
+	controllerRegistrationList := &gardencorev1beta1.ControllerRegistrationList{}
+	if err := b.K8sGardenClient.Client().List(ctx, controllerRegistrationList); err != nil {
+		return nil, err
+	}
+
+	types := make(map[string]bool)
+
+	for _, reg := range controllerRegistrationList.Items {
```

Comment:
Isn't it possible to use `ComputeRequiredExtensions` https://github.com/gardener/gardener/blob/611fb1f69673eed5f343a536b2b8ba40892872b8/pkg/operation/shoot/shoot.go#L626 to get the list of the extensions required for the shoot?

Reply:
Since `func ComputeRequiredExtensions` returns all required extensions and not only these with kind `Extension` there are several possibilities:

1. Label the namespace with all required extensions which will result in something like the following which in my opinion can be a little weird labeling.

    _backup.gardener.cloud/provider: gcp
    extensions.gardener.cloud.backupbucket/gcp: "true"
    extensions.gardener.cloud.backupentry/gcp: "true"
    extensions.gardener.cloud.controlplane/gcp: "true"
    extensions.gardener.cloud.dnsprovider/aws-route53: "true"
    extensions.gardener.cloud.extension/shoot-cert-service: "true"
    extensions.gardener.cloud.extension/shoot-oidc-service: "true"
    extensions.gardener.cloud.infrastructure/gcp: "true"
    extensions.gardener.cloud.network/calico: "true"
    extensions.gardener.cloud.operatingsystemconfig/gardenlinux: "true"
    extensions.gardener.cloud.worker/gcp: "true"
    gardener.cloud/role: shoot
    networking.shoot.gardener.cloud/provider: calico
    seed.gardener.cloud/provider: gcp
    shoot.gardener.cloud/provider: gcp_

2. Label the namespace with all required extensions of kind `Extension` using the following template `extensions.gardener.cloud/{extension-type}: "true"`. In order to not duplicate code I can filter the result of the mentioned function.

I personally tilt toward the second option but like to hear some other opinions or suggestions.
/cc @mvladev